artiq/gateware/drtio/transceiver/gtp_7series.py

```diff
@@ -696,7 +696,7 @@ def __init__(self, qpll_channel, pads, sys_clk_freq, rtio_clk_freq, mode):
         self.submodules += clock_aligner
         self.comb += [
             clock_aligner.rxdata.eq(rxdata),
-            rx_init.restart.eq(clock_aligner.restart),
+            rx_init.restart.eq(clock_aligner.restart | ~self.stable_clkin),
```

Comment:
Keeping the receiver in reset state until clock is reported stable is crucial - but I'm not sure if this is the cleanest way of doing things.